apis/v1alpha3/backendtlspolicy_types.go

```diff
@@ -126,11 +141,50 @@ type BackendTLSPolicyValidation struct {
 	// backends:
 	//
 	// 1. Hostname MUST be used as the SNI to connect to the backend (RFC 6066).
-	// 2. Hostname MUST be used for authentication and MUST match the certificate
-	//    served by the matching backend.
+	// 2. Only if SubjectAltNames is not specified, Hostname MUST be used for
+	//    authentication and MUST match the certificate served by the matching
+	//    backend.
 	//
 	// Support: Core
 	Hostname v1.PreciseHostname `json:"hostname"`
+
+	// SubjectAltNames contains one or more Subject Alternative Names.
+	// When specified, the certificate served from the backend MUST have at least one
+	// Subject Alternate Name matching one of the specified SubjectAltNames.
+	//
+	// Support: Core
+	//
+	// +optional
+	// +kubebuilder:validation:MaxItems=5
+	SubjectAltNames []SubjectAltName `json:"subjectAltNames,omitempty"`
+}
+
+// SubjectAltName represents Subject Alternative Name.
+// +kubebuilder:validation:XValidation:message="must contain Hostname, if Type is set to Hostname",rule="!(self.Type == "Hostname" && has(self.Hostname) && self.Hostname != \"\")"
+// +kubebuilder:validation:XValidation:message="must not contain Hostname, if Type is not set to Hostname",rule="self.Type != "Hostname" && has(self.Hostname) && self.Hostname != \"\""
+// +kubebuilder:validation:XValidation:message="must contain URI, if Type is set to URI",rule="!(self.Type == "URI" && has(self.URI) && self.HostName != \"\")"
+// +kubebuilder:validation:XValidation:message="must not contain URI, if Type is not set to URI",rule="self.Type != "URI" && has(self.URI) && self.URI != \"\""
+type SubjectAltName struct {
+	// Type determines the format of the Subject Alternative Name. Always required.
+	//
+	// Support: Core
+	Type SubjectAltNameType `json:"type"`
+
+	// Hostname contains Subject Alternative Name specified in DNS name format.
+	// Required when Type is set to Hostname, ignored otherwise.
+	//
+	// Support: Core
+	//
+	// +optional
+	Hostname v1.PreciseHostname `json:"hostname,omitempty"`
+
+	// URI contains Subject Alternative Name specified in URI format.
+	// Required when Type is set to URI, ignored otherwise.
+	//
+	// Support: Core
+	//
+	// +optional
+	URI string `json:"uri,omitempty"`
```

Comment:
Seems like we need some kind of regex validation here, similar in concept to what we've added for the hostname types.